.github/workflows/node.js-tests.yml

```diff
@@ -70,20 +70,25 @@ jobs:
           pnpm run build:curriculum
           pnpm run lint
 
-  test:
-    name: Test
+  # DONT REMOVE THIS JOB, you dont know what it does
```

Comment:
```suggestion
  # DONT REMOVE THIS JOB.
  # TODO: Refactor and use re-usable workflow and shared artifacts
```

I am glad you are asking to apply a plaster cast over the comment, which is already a band-aid. Jokes aside, this should be DRY with a one-build job reused across the test plane.

The issue is that we have a job defined in a different workflow, which is flaky; hence, we must duplicate it. We don't have *checks* (as in GH CI Checks), which automerging can rely on.

The flaky workflows are a different issue altogether, and adding comments about it makes it even more confusing. To be specific and laser-focused - I believe the E2E tests can never be non-flaky. No offense, adding a comment does nothing better than git blame would do, but I digress.

Instead, making the CI DRY'er is a proper fix where you do things according to their criticality. Here is how you conclude:

1. Does my code have any lint? I should stop and remove that.
2. Does my code build? I should stop and ensure it does.
3. Alright, let me do unit testing, I sure hope I am writing functional code.
4. Nice, let me put it together and test the integrity - E2E tests.

While this PR may look like it is doing a lot, It is fixing the order of the tests in the order of the user stories above and doing a bit of preventing the exclusion of wasteful runs.

As for making the CI DRY'er so that we are testing a true copy of the build, I will work on reusing build artifacts (tied to the git hash) in an upcoming PR when I have more time.

Reply:
Are you OK with the TODO above?

Reply:
> Jokes aside, this should be DRY with a one-build job reused across the test plane.

Yeah, 100%. I'll finish off the docker stuff this week so we can share images between jobs (at least in principle - I know there are more steps before we can use them).

> The issue is we don't (at least not until my hacks till yesterday) have checks (as in GH CI Checks), which automerging can rely on.

Right, but the reason we don't use the status checks is their flakiness. Anyways, it's all rendered moot by the "one-build job" approach - it's just better.

> No offense, adding a comment does nothing better than git blame would do, but I digress.

None taken. The difference is the amount of effort required to understand the context (poking through git blames is a pain). Not a big deal, though.